hack/e2e/cmd/generateServicePrincipals.go

```diff
@@ -0,0 +1,218 @@
+package cmd
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	azkeyvault "github.com/Azure/azure-sdk-for-go/services/keyvault/v7.0/keyvault"
+	"github.com/Azure/go-autorest/autorest/date"
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/jongio/azidext/go/azidext"
+	"github.com/spf13/cobra"
+
+	"github.com/Azure/ARO-RP/pkg/env"
+	msgraph_apps "github.com/Azure/ARO-RP/pkg/util/graph/graphsdk/applications"
+	msgraph_models "github.com/Azure/ARO-RP/pkg/util/graph/graphsdk/models"
+	"github.com/Azure/ARO-RP/pkg/util/graph/graphsdk/serviceprincipals"
+	"github.com/Azure/ARO-RP/pkg/util/keyvault"
```

Comment:
We may want to use the new keyvault track2 SDK instead of the old one since the keyvault SDK track2 is merged https://github.com/Azure/ARO-RP/pull/3275/files